Patch for interix, but beware of IRIX
http://bugs.gentoo.org/show_bug.cgi?id=209473

reported upstream Apr 23 2009: https://sourceforge.net/forum/message.php?msg_id=7268389

--- lzma-4.32.6.orig/src/liblzmadec/lzmadec.h	2008-06-24 09:43:40 +0200
+++ lzma-4.32.6/src/liblzmadec/lzmadec.h	2008-06-24 09:45:17 +0200
@@ -50,6 +50,12 @@
 #include <sys/types.h>
 #include <inttypes.h>
 
+#ifndef __sgi
+# ifdef HAVE_STDINT_H
+#  include <stdint.h>
+# endif
+#endif
+
 /* Define LZMADEC_NO_STDIO to not include stdio.h and lzmadec_FILE functions. */
 #ifndef LZMADEC_NO_STDIO
 #include <stdio.h>
--- lzma-4.32.6.orig/src/sdk/Common/Types.h	2008-06-24 09:43:42 +0200
+++ lzma-4.32.6/src/sdk/Common/Types.h	2008-06-24 09:47:01 +0200
@@ -9,6 +9,12 @@
 
 #include <inttypes.h>
 
+#ifndef __sgi
+# ifdef HAVE_STDINT_H
+#  include <stdint.h>
+# endif
+#endif
+
 typedef uint8_t Byte;
 typedef int32_t Int32;
 typedef uint32_t UInt32;
